tensorflow/python/training/slot_creator.py

```diff
@@ -76,11 +76,16 @@ def _create_slot_var(primary, val, scope, validate_shape, shape, dtype):
     # remove "'linear//weight' + '/'" and ':0'.
     real_slot_name = slot.name[len(primary.op.name + "/"):-2]
     slice_info = primary._save_slice_info
-    slot._set_save_slice_info(variables.Variable.SaveSliceInfo(
-        slice_info.full_name + "/" + real_slot_name,
-        slice_info.full_shape[:],
-        slice_info.var_offset[:],
-        slice_info.var_shape[:]))
+    # supprot slot's shape not same as primary's shape
+    # example: primary's shape = [10, 20, 30], slot's shape =
+    # [], [10], [10, 20] or [10, 20, 30] is allowed
+    n = slot.shape.ndims
+    if (n > 0):
```

Comment:
Why is this condition necessary? Seems like scalars should follow the same pattern.

Reply:
detail info. 
https://github.com/tensorflow/tensorflow/issues/19457
Some optimizer creates slot that should not has the same shape with primary.

Reply:
That doesn't really answer the question. Why are scalars (n == 0) treated differently than other tensors (n > 0)?

Reply:
if n==0, slot cannot slice, every slot with primary should duplicate. if n>0, slot can be slice, every slot with primary can be slice

Reply:
`lst[:0]` works fine and returns an empty list. I still don't understand the issue. If there is one, could you add a unit test which fails without the condition?